src/catalog.c

```diff
@@ -427,3 +455,55 @@ catalog_delete(Relation rel, HeapTuple tuple)
 {
 	catalog_delete_tid(rel, &tuple->t_self);
 }
+
+
+/*
+ * Invalidate TimescaleDB catalog caches.
+ *
+ * This function should be called whenever a TimescaleDB catalog table changes
+ * in a way that might invalidate associated caches. It is currently called in
+ * two distinct ways:
+ *
+ * 1. If a catalog table changes via the catalog API in catalog.c
+ * 2. Via a trigger if a SQL INSERT/UPDATE/DELETE occurs on a catalog table
+ *
+ * Since triggers (2) require full parsing, planning and execution of SQL
+ * statements, they aren't supported for simple catalog updates via (1) in
+ * native code and are therefore discouraged. Ideally, catalog updates should
+ * happen consistently via method (1) in the future, obviating the need for
+ * triggers on catalog tables that cause side effects.
+ *
+ * The invalidation event is signaled to other backends (processes) via the
+ * relcache invalidation mechanism on a dummy relation (table).
+ *
+ * Parameters: The OID of the catalog table that changed, and the operation
+ * involved (e.g., INSERT, UPDATE, DELETE).
+ */
+void
+catalog_invalidate_cache(Oid catalog_relid, CmdType operation)
+{
+	Catalog    *catalog = catalog_get();
+	CatalogTable table = catalog_table_get(catalog, catalog_relid);
+	Oid			relid;
+
+	switch (table)
+	{
+		case HYPERTABLE:
```

Comment:
The hypertable cache needs to be invalidated on inserts due to negative cache entries.

Reply:
It will be via `dimension` (you never add a hypertable entry without a dimension). On the other hand, this goes both ways so if you feel strongly about it we can treat them the same although this will generate unnecessary extra invalidation events.